src/Dotnet.Script.DependencyModel/Compilation/CompilationDependencyResolver.cs

```diff
@@ -1,28 +1,62 @@
+        private readonly IRestorer _restorer;
+
+        public CompilationDependencyResolver(LogFactory logFactory) : this(new ScriptProjectProvider(logFactory), new ScriptDependencyContextReader(logFactory), new CompilationReferencesReader(logFactory), logFactory)
         {
         }
 
-        public CompilationDependencyResolver(ScriptProjectProvider scriptProjectProvider, ICompilationReferenceReader compilationReferenceReader, LogFactory logFactory)
+        public CompilationDependencyResolver(ScriptProjectProvider scriptProjectProvider, ScriptDependencyContextReader scriptDependencyContextReader, ICompilationReferenceReader compilationReferenceReader, LogFactory logFactory)
         {
             _scriptProjectProvider = scriptProjectProvider;
+            this._scriptDependencyContextReader = scriptDependencyContextReader;
             _compilationReferenceReader = compilationReferenceReader;
+            _restorer = CreateRestorer(logFactory);
         }
 
-        public IEnumerable<CompilationReference> GetDependencies(string targetDirectory, IEnumerable<string> scriptFiles, bool enableScriptNugetReferences, string defaultTargetFramework = "net46")
+        public IEnumerable<CompilationDependency> GetDependencies(string targetDirectory, IEnumerable<string> scriptFiles, bool enableScriptNugetReferences, string defaultTargetFramework = "net46")
         {
             var projectFileInfo = _scriptProjectProvider.CreateProject(targetDirectory, scriptFiles, defaultTargetFramework, enableScriptNugetReferences);
-            return _compilationReferenceReader.Read(projectFileInfo);
+            _restorer.Restore(projectFileInfo, packageSources: Array.Empty<string>());
+            var pathToAssetsFile = Path.Combine(Path.GetDirectoryName(projectFileInfo.Path), "obj", "project.assets.json");
+            var dependencyContext = _scriptDependencyContextReader.ReadDependencyContext(pathToAssetsFile);
+            var result = new List<CompilationDependency>();
+            foreach (var scriptDependency in dependencyContext.Dependencies)
+            {
+                var compilationDependency = new CompilationDependency(scriptDependency.Name, scriptDependency.Version, scriptDependency.CompileTimeDependencyPaths, scriptDependency.ScriptPaths);
+                result.Add(compilationDependency);
+            }
+
+            // On .Net Core, we need to fetch the compilation references for framework assemblies separately.
+            if (defaultTargetFramework.StartsWith("netcoreapp3", StringComparison.InvariantCultureIgnoreCase))
+            {
+                var compilationreferences = _compilationReferenceReader.Read(projectFileInfo);
+                result.Add(new CompilationDependency("Microsoft.NETCore.App", ScriptEnvironment.Default.NetCoreVersion.Version, compilationreferences.Select(cr => cr.Path).ToArray(), Array.Empty<string>()));
```

Comment:
does this actually work? if `ScriptEnvironment.Default.NetCoreVersion.Version` is invoked from omnisharp, it will throw, no?

Reply:
No, we don't throw  anymore. It comes back as "unknown". It does not really matter what the version number is. It is not used for anything. Lets just put `3.0` instead 👍